lib/install/validate/config.go

```diff
@@ -25,106 +26,322 @@ import (
-			log.Warningf("  %q", h)
+		for _, h := range status.MisconfiguredDisabled {
+			log.Warningf("\t%q", h)
+		}
+		log.Infof("Firewall must permit %s %d/%s %s to VCH management interface if firewall is reenabled",
+			status.Rule.PortType, status.Rule.Port, status.Rule.Protocol, status.Rule.Direction)
+	}
+
+	preMsg := "Firewall allowed IP configuration may prevent required connection on hosts:"
+	postMsg := fmt.Sprintf("Firewall must permit %s %d/%s %s to the VCH management interface",
+		status.Rule.PortType, status.Rule.Port, status.Rule.Protocol, status.Rule.Direction)
+	v.FirewallCheckDHCPMessage(status.UnknownEnabled, preMsg, postMsg)
+
+	preMsg = "Firewall configuration may be incorrect if firewall is reenabled on hosts:"
+	postMsg = fmt.Sprintf("Firewall must permit %s %d/%s %s to the VCH management interface if firewall is reenabled",
+		status.Rule.PortType, status.Rule.Port, status.Rule.Protocol, status.Rule.Direction)
+	v.FirewallCheckDHCPMessage(status.UnknownDisabled, preMsg, postMsg)
+}
+
+// FirewallCheckDHCPMessage outputs warning message when we are unable to check
+// that the management interface is allowed by the host firewall allowed IP rules due to DHCP
+func (v *Validator) FirewallCheckDHCPMessage(hosts []string, preMsg, postMsg string) {
+	if len(hosts) > 0 {
+		log.Warn("Unable to fully verify firewall configuration due to DHCP use on management network")
+		log.Warn("VCH management interface IP assigned by DHCP must be permitted by allowed IP settings")
+		log.Warn(preMsg)
+		for _, h := range hosts {
+			log.Warningf("\t%q", h)
+		}
+		log.Infof(postMsg)
+	}
+}
+
+// CheckIPInNets checks that an IP is within allowedIPs or allowedNets
+func (v *Validator) CheckIPInNets(checkIP net.IPNet, allowedIPs []string, allowedNets []string) bool {
+	for _, a := range allowedIPs {
+		aIP := net.ParseIP(a)
+		if aIP != nil && checkIP.IP.Equal(aIP) {
+			return true
 		}
-		log.Warnf("Firewall must permit %d/tcp outbound if firewall is reenabled", rule.Port)
 	}
-	return
+
+	for _, n := range allowedNets {
+		_, aNet, err := net.ParseCIDR(n)
+		if err != nil {
+			continue
+		}
+		if aNet.Contains(checkIP.IP) {
+			return true
+		}
+	}
+	return false
+}
+
+// FirewallEnabled checks if the host firewall is enabled
+func (v *Validator) FirewallEnabled(host *object.HostSystem) (bool, error) {
+	esxfw, err := esxcli.GetFirewallInfo(host)
+	if err != nil {
+		return false, err
+	}
+	if esxfw.Enabled {
+		log.Infof("Firewall status: ENABLED on %q", host.InventoryPath)
+		return true, nil
+	}
+	log.Infof("Firewall status: DISABLED on %q", host.InventoryPath)
+	return false, nil
+}
+
+// GetMgmtIP finds the management network IP in config
+func (v *Validator) GetMgmtIP(conf *config.VirtualContainerHostConfigSpec) net.IPNet {
+	var mgmtIP net.IPNet
+	if conf != nil {
+		for _, n := range conf.ExecutorConfig.Networks {
+			if n.Network.Common.Name == "management" {
+				if n.IP != nil {
+					mgmtIP = *n.IP
+					return mgmtIP
+				}
```

Comment:
if the management network is sharing with external or client network, and there is static ip, can this method find the mgmt ip address?